Add method to properly decode the Transaction Extra Field

KosnIre opened the pull request:
## Status
TESTING, review comments ok

## Summary
In addition to creating `decode_tx_extra()` and logic as per the whitepaper, I have also done a little restructuring in the file to support offline testing (not connected to Discord server). It takes effect if you don't have a Token set. This way, I was able to just print results to the command line.

My test file is isolated further and does not use Turtlecoind either.

**_IMPORTANT_:** The changes include APIs calls to the _latest_ (development) version of `turtlecoin` which is not released yet. I have installed that module from source. That is why there is some extra churn in the diff. I might be able to revert some of the changes if needed.

## Changes
- Added `decode_tx_extra()` method, which accepts a tx extra hex string and returns a string of decoded data. This keeps a marker (`curr_index`) on the hex string and moves along it checking for subfields. Currently it tries to decode custom data in UTF-8 format. It may be possible to add ascii as a fallback.
- Turtlecoind connection is now established in `connect_to_turtlecoind()`. This only occurs if the file is being run (as opposed to imported), i.e. `if __name__ == '__main__':`
- Updated to latest `turtlecoin` code. Notice APIs are now snake_case and return a `['result']` dictionary.
- Fixed a few minor lint issues.

Sajo811 commented:
Update: reviewing it right now, and fixing some issues with Taco on Discord. Wanna join test server? https://discord.gg/xkP5AAS

Sajo811 commented:
#2

Sajo811 commented:
dang this reviewing this is lame

KosnIre commented:
Ok, I made some new changes to parse the tx_extra field out of every transaction in the block.
I changed the `prettyPrintStats()` method to break out each transaction on its own, instead of printing the arrays. I think this will read better.

I also added a unit test file for `prettyPrintStats()`. It's very basic right now, but it helped to make sure I was printing what I expected.
